chat-widget/src/components/webchatcontainerstateful/webchatcontroller/middlewares/renderingmiddlewares/messageSequenceIdOverrideMiddleware.tsx

```diff
@@ -0,0 +1,66 @@
+import { IWebChatAction } from "../../../interfaces/IWebChatAction";
+import { WebChatActionType } from "../../enums/WebChatActionType";
+import { Constants } from "../../../../../common/Constants";
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/no-unused-vars
+const createMessageSequenceIdOverrideMiddleware = ({ dispatch }: { dispatch: any }) => (next: any) => (action: IWebChatAction) => {
+    if (isApplicable(action)) {
+        return next(overrideSequenceIdWithOriginalMessageId(action));
+    }
+    return next(action);
+};
+
+const isApplicable = (action: IWebChatAction): boolean => {
+
+    return action.type === WebChatActionType.DIRECT_LINE_INCOMING_ACTIVITY &&
+        isValidChannel(action) &&
+        lookupOriginalMessageId(action) !== undefined &&
+        isWebSequenceIdPresent(action);
+};
+
+const isValidChannel = (action: IWebChatAction): boolean => {
```

Comment:
I remember there are other middlewares using similar logics. Can we extract them into a common place, so that we can add unit test on them?

Reply:
the other middleware (createMessageTimeStampMiddleware) using it will be removed in a posterior task since is not in use